rca/guides/models.py

```diff
@@ -125,11 +128,46 @@ def get_related_pages(self):
             )
         return related_pages
 
+    def has_sticky_cta(self):
+        data = self.get_sticky_cta()
+        return all(data.get(key) for key in ["message", "action", "link"])
+
+    def clean(self, *args, **kwargs):
+        super().clean(*args, **kwargs)
+
+        if self.sticky_cta_page and self.sticky_cta_link_url:
+            message = "You must specify a page or link url. You can't use both."
+            raise ValidationError(
+                {
+                    "sticky_cta_link_url": ValidationError(message),
+                    "sticky_cta_page": ValidationError(message),
+                }
+            )
+
+        if bool(self.sticky_cta_link_url) != bool(self.sticky_cta_link_text):
+            raise ValidationError(
+                {
+                    "sticky_cta_link_text": ValidationError(
+                        "You must specify link text, if you use the link url field."
+                    )
+                }
+            )
+
+        if self.sticky_cta_page or self.sticky_cta_link_url:
+            # Either page or link URL is specified, so description is required
+            if not self.sticky_cta_description:
+                message = "You must specify a description for the sticky CTA"
+                raise ValidationError(
+                    {"sticky_cta_description": ValidationError(message)}
+                )
+
```

Comment:
**thought**

As this is a mixing, is it possible to define the validation on the StickyCTA class itself then call super().clean ?

Reply:
I initially did this by overriding the `clean()` method on the StickyCTA class, but it never worked.

However, in 3ec923c3 I have added the validation in the `full_clean()` method and this is working